patches/api/0036-Add-PlayerUseUnknownEntityEvent.patch

```diff
@@ -3,56 +3,96 @@ From: Jedediah Smith <jedediah@silencegreys.com>
 --- /dev/null
 +++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerUseUnknownEntityEvent.java
-@@ -0,0 +1,46 @@
+@@ -0,0 +1,82 @@
 +package com.destroystokyo.paper.event.player;
 +
++import io.papermc.paper.math.Position;
 +import org.bukkit.entity.Player;
 +import org.bukkit.event.HandlerList;
 +import org.bukkit.event.player.PlayerEvent;
 +import org.bukkit.inventory.EquipmentSlot;
 +import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
 +
++/**
++ * Represents an event that is called when a player right-clicks an unknown entity.
++ * Useful for plugins dealing with virtual entities (entities that aren't actually spawned on the server).
++ * <br>
++ * This event may be called multiple times per interaction with different interaction hands
++ * and with or without the clicked position.
++ */
 +public class PlayerUseUnknownEntityEvent extends PlayerEvent {
 +
-+    private static final HandlerList handlers = new HandlerList();
++    private static final HandlerList HANDLERS = new HandlerList();
 +    private final int entityId;
 +    private final boolean attack;
-+    @NotNull private final EquipmentSlot hand;
++    private final @NotNull EquipmentSlot hand;
++    private final @Nullable Position clickedPosition;
 +
-+    public PlayerUseUnknownEntityEvent(@NotNull Player who, int entityId, boolean attack, @NotNull EquipmentSlot hand) {
++    public PlayerUseUnknownEntityEvent(@NotNull Player who, int entityId, boolean attack, @NotNull EquipmentSlot hand, @Nullable Position clickedPosition) {
 +        super(who);
 +        this.entityId = entityId;
 +        this.attack = attack;
 +        this.hand = hand;
++        this.clickedPosition = clickedPosition;
 +    }
 +
++    /**
++     * Returns the entity id of the unknown entity that was interacted with.
++     *
++     * @return the entity id of the entity that was interacted with
++     */
 +    public int getEntityId() {
 +        return this.entityId;
 +    }
 +
++    /**
++     * Returns whether the interaction was an attack.
++     *
++     * @return true if the player is attacking the entity, false if the player is interacting with the entity
++     */
 +    public boolean isAttack() {
 +        return this.attack;
 +    }
 +
-+    @NotNull
-+    public EquipmentSlot getHand() {
++    /**
++     * Returns the hand used to perform this interaction.
++     *
++     * @return the hand used to interact
++     */
++    public @NotNull EquipmentSlot getHand() {
 +        return this.hand;
 +    }
 +
++    /**
++     * Returns the exact position that was clicked, or null if not available.
++     * See {@link org.bukkit.event.player.PlayerInteractAtEntityEvent} for more details.
++     *
++     * @return the exact position that was clicked, or null if not available
++     * @see org.bukkit.event.player.PlayerInteractAtEntityEvent
++     */
++    public @Nullable Position getClickedPosition() {
```

Comment:
I think the vector in the packet is a relative vector to the armor stand, it's not a position.